s3_management/manage.py

```diff
@@ -121,8 +120,8 @@ def between_bad_dates(package_build_time: datetime):
 
 
 class S3Index:
-    def __init__(self: S3IndexType, objects: List[str], prefix: str) -> None:
-        self.objects = objects
+    def __init__(self: S3IndexType, objects: Dict[str, str], prefix: str) -> None:
```

Comment:
The RHS should be `str | None`

```suggestion
    def __init__(self: S3IndexType, objects: Dict[str, str | None], prefix: str) -> None:
```

Reply:
I think there's a couple other places this should be done as well.